tools/sense_studio/static/video_recording.js

```diff
@@ -0,0 +1,137 @@
+}
+
+
+function displayOverlay(text, mode) {
+    overlay = document.getElementById('textOverlay');
+    container = document.getElementById('videoContainer');
+
+    // Update overlay text
+    overlay.innerHTML = text;
+
+    // Update overlay visibility
+    if (text !== '') {
+        overlay.classList.remove('hidden');
+    } else {
+        overlay.classList.add('hidden');
+    }
+
+    // Update overlay color
+    if (mode === 'saved') {
+        overlay.classList.add('green');
+        overlay.classList.remove('red');
+    } else if (mode === 'error') {
+        overlay.classList.remove('green');
+        overlay.classList.add('red');
+    } else {
+        overlay.classList.remove('green');
+        overlay.classList.remove('red');
+    }
+
+    // Update container color
+    if (mode === 'recording') {
+        container.classList.add('red');
+        container.classList.add('inverted');
+    } else {
+        container.classList.remove('red');
+        container.classList.remove('inverted');
+    }
+}
+
+
+function setupRecording(stream, url) {
+    let player = document.getElementById('player');
+    player.srcObject = stream;
+
+    countdownDuration = parseInt(document.getElementById('countdown').value)
+    recordingDuration = parseInt(document.getElementById('duration').value)
+
+    // Show countdown
+    for (const seconds of Array(countdownDuration).keys()) {
+        const countdown = countdownDuration - seconds;
+        setTimeout(displayOverlay, seconds * 1000, 'Get Ready: ' + countdown, 'countdown');
+    }
+
+    // Start recording
+    setTimeout(startRecording, countdownDuration * 1000, stream, recordingDuration, url);
+};
+
+
+function startRecording(stream, recordingDuration, url) {
+    mediaRecorder = new MediaRecorder(stream, {mimeType: 'video/webm; codecs=vp8'});
+    mediaRecorder.ondataavailable = function (event) {
+        if (event.data.size > 0) {
+            saveVideo(event.data, url);
+        }
+    };
+    mediaRecorder.start();
+
+    // Show countdown
+    for (const seconds of Array(recordingDuration).keys()) {
+        const countdown = recordingDuration - seconds;
+        setTimeout(displayOverlay, seconds * 1000, countdown, 'recording');
+    }
+
+    // Stop recording
+    setTimeout(stopRecording, recordingDuration * 1000, mediaRecorder);
+}
+
+
+function stopRecording(mediaRecorder) {
+    mediaRecorder.stop();
+    mediaRecorder.stream.getTracks().forEach(function(track) {
+        track.stop();
+    });
+
+    displayOverlay('', false);
+}
+
+
+function saveVideo(chunk, url) {
+    let blob = new Blob([chunk], {type: "video/webm"});
+    const formData = new FormData();
+    formData.append('video', blob);
+    fetch(url, {
+        method: 'POST',
+        body: formData,
+    }).then(res => {
+        if (res.ok) {
+            displayOverlay('Video Saved', 'saved');
+        } else {
+            displayOverlay('Error: ' + res.status, 'error');
```

Comment:
Better style is to use a template literal: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Reply:
Nice, there's the f-string I was looking for :grin: